modules/k6/k6.go

```diff
@@ -49,13 +54,27 @@ func SetEnvVar(variable string, value string) testcontainers.CustomizeRequestOpt
 	}
 }
 
-// WithCache uses the given volume as a cache for building the k6 binary.
-// If the volume does not exists, it is created.
-func WithCache(cache string) testcontainers.CustomizeRequestOption {
+// WithCache sets a volume as a cache for building the k6 binary
+// If a volume name is provided in the TC_K6_BUILD_CACHE, this volume is used.
+// If no value is provided, a volume is created and removed when the test session ends.
+func WithCache() testcontainers.CustomizeRequestOption {
+	var volOptions *mount.VolumeOptions
+
+	cacheVol := os.Getenv("TC_K6_BUILD_CACHE")
+	// if no volume is provided, create one and ensure add labels for garbage collection
+	if cacheVol == "" {
+		sessionID := testcontainerssession.SessionID()
+		cacheVol = fmt.Sprintf("k6-cache-%s", sessionID)
+		volOptions = &mount.VolumeOptions{
+			Labels: testcontainersdocker.DefaultLabels(sessionID),
```

Comment:
I'm going to merge this today: https://github.com/testcontainers/testcontainers-go/pull/1790/files#diff-e9b78e9b34e14f076ba250ebd1042b315e63673569f82372a8064cece5f034bfR211

It includes a function that exposes the labels out of the internal package. Could you update this PR once the other is merged 🙏 ?

Reply:
Sure please ping me when it is merged.

Reply:
Done!

Reply:
@mdelapenya Something I don't see is how to get the `SessionID` if I don't use the internal package. I need it to create a unique name for the cache volume for each session. I cannot use a simple `UUID` because I need to ensure all tests in a session use the same volume name.

I could get it from the `sessionId` label in the `GenericLabels` map, but the name of this label is defined in the `testcontainersdocker` internal package. 

Maybe it should be exposed in the `testcontainers` package? Notice that at this point the container is not created and therefore I don't have access to `Container.SessionID`

Reply:
Isn't it enough with this? The sessionID is an implementation detail of the internals of the library.

```go
Labels: testcontainers.GenericLabels()
```
But I see you point regarding the cacheVol variable. You could get a hash of the labels map, so we do not need to expose anything else at the core. I would see the value in exposing the SessionID, but at the moment there is a way to get a unique identifier for that volume, formed by the hash of:
```go
return map[string]string{
		LabelBase:      "true",
		LabelLang:      "go",
		LabelSessionID: sessionID,
		LabelVersion:   internal.Version,
	}
```
Probably more work calculating the hash at the consumer side, that's true, at the cost of saving another method in the interface. If we see that consuming the SessionID is useful in more than one use case, then we can for sure add it back, not affecting this code, as how the vol is names is another implementation detail.

In any case, I'm not opposed to expose it in a separate PR so that this consumes it, but I'd like to get a clear value out of it. Thoughts?

Reply:
I find it a little bit obscure and fragile to use a hash of the labels. I need this hash to be the same for all tests in a session. Therefore, all elements of the map should be the same for all tests, which seems to be the case presently, but this is an implementation detail that can change at any moment.

I'm curious about your reluctance to expose the SessionID when it is already exposed in `Container` interface. The problem is that I don't have access to it until the container is created and this logic is executed before this.

Reply:
I see.. I did not remember the `container.SessionID` method tbh, thanks for pointing out to that. Let's do it as you propose: I'll send a PR today exposing the SessionID as part of the core so that consumers can read it.